File: leetcode-392-Is_Subsequence/leetcode-392-Is_Subsequence.cpp
#include <cstdlib>
#include <iostream>
#include <string>

bool isSubsequence(std::string s, std::string t) {
    size_t t_i = 0;
    bool is_apear = (s.size() && t.size()) || (!s.size());
    for (size_t s_i = 0; s_i < s.size() && t_i < t.size() && is_apear ; ++s_i)
    {
        is_apear = false;
        for (; t_i < t.size() && !is_apear ; ++t_i)
            is_apear = (s[s_i] == t[t_i]);

        if(t_i == t.size() && s_i != s.size()-1)
            is_apear = false;
    }
    return is_apear;
}

int main(){
    // std::string s = "b",
    //             t = "c";
    std::string s = "abc",
                t = "ahbgdc";
    std::cout << isSubsequence(s, t) << std::endl;
    return EXIT_SUCCESS;
}

File: leetcode-392-Is_Subsequence/leetcode-392-Is_Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        j:int = -1
        for i in range(len(s)):
            for j in range(j+1, len(t)):
                if t[j] == s[i]:
                    break
            else:
                return False
        return True


if __name__ == "__main__":
    s:str = "abc"
    t:str = "ahbgdc"

    res:bool = Solution().isSubsequence(s, t)

    print(res)